src/masternodes/mn_rpc.cpp

```diff
@@ -2896,6 +2977,29 @@ UniValue listaccounthistory(const JSONRPCRequest& request) {
             if (owner != startKey.owner || (height > startKey.blockHeight || (depth <= startKey.blockHeight && (height < startKey.blockHeight - depth))))
                 return false;
 
+            if(!tokenFilter.empty()) {
+                bool hasToken = false;
+                for (auto const & diff : diffs) {
+                    auto token = pcustomcsview->GetToken(diff.first);
+                    std::string const tokenIdStr = token->CreateSymbolKey(diff.first);
+
+                    if(tokenIdStr == tokenFilter) {
+                        hasToken = true;
+                        break;
+                    }
+                }
+
+                if(!hasToken) {
+                    return true; // continue
+                }
+            }
+
+            if(noRewards) {
+                if(category == static_cast<unsigned char>(CustomTxType::NonTxRewards)) {
+                    return true; // continue
+                }
+            }
```

Comment:
It looks like the same code repeated 3 times, can put to a function?

Reply:
I think we can refactor it, later, are you OK? I mean he want to ship fix ASAP.

Reply:
Sounds good